### PR TITLE
Add mip store bootstrap

### PR DESCRIPTION
**Pull request by sydhds**

### CODE REVIEW

File: massa-proto/build.rs [hunk 1]
```diff
@@ -7,6 +7,9 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     //TODO add download external protos files instead of doing it manually
     let protos = find_protos("proto/massa/");
 
+    let mut prost_config = prost_build::Config::default();
+    prost_config.protoc_arg("--experimental_allow_proto3_optional");
```
Comment (Leo-Besancon) [inline massa-proto/build.rs:11]: Are these changes still necessary with https://github.com/massalabs/massa/pull/3764/files merged?

File: massa-node/src/main.rs [hunk 2]
```diff
@@ -339,7 +341,13 @@ async fn launch(
         block_count_considered: MIP_STORE_STATS_BLOCK_CONSIDERED,
         counters_max: MIP_STORE_STATS_COUNTERS_MAX,
     };
-    let mip_store = MipStore::try_from(([], mip_stats_config)).unwrap();
+    let mut mip_store =
+        MipStore::try_from(([], mip_stats_config)).expect("Cannot create an empty MIP store");
```
Comment (Leo-Besancon) [inline massa-node/src/main.rs:345]: I'm not sure I remember the design choice here. When we want to add a MIP, the devs would just include it in the store init here?
And after that the update_with below will check compatibilities between our MIP store and the MIP store of our bootstrap server?

**Review by Leo-Besancon — COMMENTED** [review]
It seems clean to me👍. I think it would be worth it to have a second opinion (maybe @Ben-PH?) on the bootstrap module additions, in case I missed something there!

**Review by Ben-PH — APPROVED** [review]
This shouldn't cause any issues with the bootstrap system from what I can tell.

File: massa-node/src/main.rs (hunk 2 above)
Comment (Leo-Besancon) [reply massa-node/src/main.rs:345]: Ok!

**Review by Leo-Besancon — APPROVED** [review]
Nice, I think Versioning is coming together nicely!
